web/src/fileview/fileview.js

```diff
@@ -211,6 +211,138 @@ function init(initData) {
     });
   }
 
+  // compute the offset from the start of the parent containing the click
+  function textBeforeOffset(childNode, childOffset, parentNode) {
+    // create a new range starting at the beginning of the parent and going until the selection
+    const rangeBeforeClick = new Range();
+    rangeBeforeClick.setStart(parentNode, 0);
+    rangeBeforeClick.setEnd(childNode, childOffset);
+    return rangeBeforeClick.toString();
+  }
+
+  // returns range for symbol containing the specified location.
+  // symbol is determined by greedily absorbing alphanumerics and underscores.
+  function symbolAtLocation(textNode, offset) {
+    const stringBefore = textNode.nodeValue.substring(0, offset);
+    const stringAfter = textNode.nodeValue.substring(offset);
+    const startIndex = stringBefore.match(/[a-zA-Z0-9_]*$/).index;
+    const endIndex = stringBefore.length + stringAfter.match(/^[a-zA-Z0-9_]*/)[0].length;
+    const range = new Range();
+    range.setStart(textNode, startIndex);
+    range.setEnd(textNode, endIndex);
+    return range;
+  }
+
+  function triggerJumpToDef(event) {
+      const nodeClicked = document.getSelection().anchorNode.parentNode;
+      const cachedUrl = nodeClicked.getAttribute('definition-url');
+      if (cachedUrl) {
+        window.location.href = cachedUrl;
+      }
+  }
+
+  var hoveringNode = null;
+  var mousePositionX = 0;
+  var mousePositionY = 0;
+
+  function cancelHover() {
+    if (hoveringNode) {
+      hoveringNode.className = 'hoverable';
+    }
+    hoveringNode = null;
+  }
+
+  function hoverOverNode(node) {
+    node.className = 'hovering';
+    hoveringNode = node;
+  }
+
+  function checkIfHoverable(node) {
+    var info = getFileInfo();
```

Comment:
Can you use initData here for getting the repoName instead of duplicating the URL parsing logic in JS?

(the assumption that the repo name doesn't have slashes in it is also broken for GitHub repositories)